src/templates/tag.js

```diff
@@ -49,52 +45,87 @@ function getTagDependencies(tagStruct, tags) {
   }));
 }
 
-module.exports = (page, metaIndex) => {
-  const tagClassSnake = page._pathParts[page._pathParts.length - 1];
-  const tagClassPascal = page.tagClass || tagClassSnake
+function getChildTags(tagStruct, tags) {
+  return Object.values(tags).filter(otherTagStruct =>
+    //Invader's BasicObject is the only inherited struct which is not a traditional tag type
+    otherTagStruct.name != "BasicObject" && (
+      otherTagStruct.inherits == tagStruct.name ||
+      tagStruct.name == "Object" && otherTagStruct.inherits == "BasicObject"
+    )
+  );
+}
```

Comment:
In Invader's tag definitions, the Placeholder and Scenery structs inherit from BasicObject which in turn inherits from Object. The hacky workaround here ensures that "scenery" is listed as a child on the "object" page. This "BasicObject" may be some kind of structure used to represent immobile objects, but I don't think it's an abstract tag itself so I guess it needs to be treated the same as "Object".

In general, Invader's struct definitions are difficult to reconcile with the pages we have because of differences in naming conventions, the lack of a flag for structs indicating if they represent a canonical tag class, and quirks like this. It's not really a fault of the definitions because they serve their intended use case, but I think adding some more structured data linking canonical tags with invader's structs would help simplify code. That could be done in a few places:

1. By updating Invader's tag definitions to help suit our needs, if Kava's open to that
2. Add some additional data to YAML prefixes in c20 to help link pages with struct names
3. A JSON or JS file in this project which ties together Invader struct names with tag classes

I don't really want to solve that problem in this PR because it's quite open-ended. I'd rather keep the diff small and tackle it later.

Reply:
I'm pretty quick to jump onto 2 or 3.